linode/lkenodepool/framework_resource.go

```diff
@@ -0,0 +1,269 @@
+				IDType: types.StringType,
+				Schema: &resourceSchema,
+			},
+		),
+	}
+}
+
+type Resource struct {
+	helper.BaseResource
+}
+
+func (r *Resource) Read(
+	ctx context.Context,
+	req resource.ReadRequest,
+	resp *resource.ReadResponse,
+) {
+	tflog.Debug(ctx, "Read linode_lke_node_pool")
+	var data NodePoolModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if helper.FrameworkAttemptRemoveResourceForEmptyID(ctx, data.ID, resp) {
+		return
+	}
+
+	ctx = helper.SetLogFieldBulk(ctx, map[string]any{
+		"cluster_id": data.ClusterID.ValueInt64(),
+		"pool_id":    data.ID.ValueString(),
+	})
+
+	clusterID, poolID := data.ExtractClusterAndNodePoolIDs(&resp.Diagnostics)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Trace(ctx, "client.GetLKENodePool(...)")
+	nodePool, err := client.GetLKENodePool(ctx, clusterID, poolID)
+	if err != nil {
+		if lerr, ok := err.(*linodego.Error); ok && lerr.Code == 404 {
+			resp.Diagnostics.AddWarning(
+				"Error reading Linode Node Pool",
+				fmt.Sprintf("Removing Linode Node Pool %d, cluster %d, from state because it no longer exists", poolID, clusterID),
+			)
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Error reading Linode Node Pool %d, cluster %d", poolID, clusterID),
+			err.Error(),
+		)
+		return
+	}
+
+	data.FlattenLKENodePool(ctx, clusterID, nodePool, false, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+	tflog.Trace(ctx, "Read linode_lke_node_pool done")
+}
+
+func (r *Resource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	tflog.Debug(ctx, "Create linode_lke_node_pool")
+	var data NodePoolModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var createOpts linodego.LKENodePoolCreateOptions
+
+	data.SetNodePoolCreateOptions(ctx, &createOpts, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	clusterID := helper.FrameworkSafeInt64ToInt(
+		data.ClusterID.ValueInt64(),
+		&resp.Diagnostics,
+	)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, "client.CreateLKENodePool(...)", map[string]any{
+		"cluster_id": clusterID,
+		"options":    createOpts,
+	})
+	pool, err := client.CreateLKENodePool(ctx, clusterID, createOpts)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating Linode Node Pool", err.Error())
+		return
+	}
+
+	tflog.Debug(ctx, "waiting for node pool to enter ready status")
+	readyPool, err := WaitForNodePoolReady(ctx,
+		*client,
+		int(r.Meta.Config.EventPollMilliseconds.ValueInt64()),
+		clusterID,
+		pool.ID,
+	)
+	if err != nil {
+		resp.Diagnostics.AddError("Linode Node Pool is not ready after create", err.Error())
+		return
+	}
+
```

Comment:
One last thing, overwise this should be good to go!

Could you explicitly set the ID here (like we do [here](https://github.com/linode/terraform-provider-linode/blob/dev/linode/sshkey/framework_resource.go#L69))?

This is necessary because in some edge cases the ID is pre-populated with an empty string, which then won't be overridden by the flatten function if preserveKnown is true.

Reply:
done